docs/how-it-works.md

```diff
@@ -0,0 +1,54 @@
+# How it works
+
+The objective of this document explains how the NGINX Ingress controllers works, in particular how the NGINX model is built and why we need a one.
+
+## NGINX configuration
+
+The goal of this Ingress controller is the assembly of a configuration file (nginx.conf). This is something we need to do, at least until [this PR is merged in OpenResty](0). The main implication of this requirement is the need to reload NGINX after a change in the configuration file.
```

Comment:
We can probably exclude:

```
This is something we need to do, at least until [this PR is merged in OpenResty](0). The main implication of this requirement is the need to reload NGINX after a change in the configuration file.
```

since, the Ingress Controller is always going to assemble a config file, in any case. We can add that PR to another section detailing what it does.